extensions/storage/README.md

```diff
@@ -0,0 +1,78 @@
+# Object Storage Extension Specification
+
+- **Title: Object Storage**
+- **Identifier: storage**
+- **Field Name Prefix: -**
+- **Scope: Item, Assets**
+- **Extension [Maturity Classification](../README.md#extension-maturity): Proposal**
+- **Owner**: @davidraleigh
+
+This document explains the fields of the Object Storage Extension to a STAC Item and an Asset. This does not cover NFS solutions provided by PaaS cloud companies. Allows user to specify details related to cloud storage access and costs for assets and their STAC item.
+
+- [Example (Landsat 8)](examples/example-landsat8.json)
+- [JSON Schema](json-schema/schema.json)
+
+## Asset fields
+
+| Field Name  | Type   | Description |
+| ----------- | ------ | ----------- |
+| storage:cloud_platform        | string    | The [cloud provider](#providers) where data is stored |
+| storage:manager               | string    | The entity in charge of managing the data. |
+| storage:regions               | [string]  | The region(s) where the data is stored. Relevant to speed of access and inter region egress costs (as defined by PaaS provider) |
+| storage:bucket                | string    | The bucket for the asset, used along with object path |
+| storage:object_path           | string    | The object_path for the asset, used along with bucket |
```

Comment:
Maybe just `path`?

Reply:
> Maybe just path?

I think `storage:container` and `storage:path` are good names. With an explicit example, e.g., 

```
{ 
 "href": "https://foobar-data.s3.us-west-2.amazonaws.com/some/path/to/a/cog.tif",
  "storage:platform": "aws_s3",
  "storage:container": "foobar-data",
  "storage:path": "some/path/to/a/cog.tif",
  "storage:region":"us-west-2"
}
```

Reply:
> I think `storage:container` and `storage:path` are good names. With an explicit example, e.g.,
It would be strange to adopt Azure's `container` when all the other providers use the term `bucket`

`key` is a more common term used by cloud providers (in addition to `object` and `blob`). `path` isn't really anywhere in documentation. I just added it to `object` to make that more clear.

Reply:
key is more descriptive as well, I think it's a better term than path

Reply:
I'm fine either way as long as we avoid the `_` 😅. For people not so much into cloud storage path seems more intuitive, but if key is an accepted term across providers, then I guess it's fine.

Reply:
bucket and key sound fine then, if all the other providers use it. I was trying to counter my AWS-bias and overcorrected.